pkg/kubelet/kubelet_pods.go

```diff
@@ -487,6 +493,22 @@ func managedHostsFileContent(hostIPs []string, hostName, hostDomainName string,
 	return buffer.Bytes()
 }
 
+// isSupportIPv6 Determine whether the system supports ipv6 by checking
+// loopback interface inside the pod whether actually assigned IPv6 IP.
+func isSupportIPv6() bool {
+	loopbackIP, err := netutil.ChooseIpv6IPForInterface(netutil.LoopbackInterfaceName)
```

Comment:
This is looking at the _host_'s `lo`, not the pod's.

Hm... I guess kubelet doesn't actually know about the pod loopback interface; CRI only tells it about the primary NIC.

Huh. OK, maybe we _should_ just check the global ipv6 support sysctl.

Reply:
> This is looking at the _host_'s `lo`, not the pod's.

This was a stupid comment and I mistakenly thought I had fixed it to be correct.



> Huh. OK, maybe we _should_ just check the global ipv6 support sysctl.

Just questioning, if we fail to get an ipv6 IP on the host's loopback interface, isn't that enough to verify that the global sysctl has disable IPv6?

Reply:
In theory, the user could have disabled IPv6 on just that interface (ie, `net.ipv6.conf.lo.disable_ipv6 = 1`, `net.ipv6.conf.default.disable_ipv6 = 0`. That's why I wanted to base it on the actual loopback interface in the pod.

If we're _not_ basing it on whether the pod _actually_ has `::1`, then there are going to be edge cases we don't get right. (eg, the user could have IPv6 enabled in the host netns but be disabling it in pod netns for some reason). And if we can't get it _exactly right_, we might as well do something simple and mostly-right, and just testing the global sysctl is simpler than checking the host netns `lo` interface.

(Maybe if we already had all the helper functions to be able to test `lo` properly then that would be an OK solution, but I don't like adding new helper functions which we will have to support going forward, just to do something that isn't quite right anyway.)